docs/adr/0008-managementv3-clusters-support.md

```diff
@@ -0,0 +1,49 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [8. Management V3 clusters support](#8-management-v3-clusters-support)
+  - [Context](#context)
+  - [Decision](#decision)
+  - [Consequences](#consequences)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# 8. Management V3 clusters support
+
+* Status: proposed
+* Date: 2024-01-25
+* Authors: @salasberryfin
+* Deciders: @richardcase @alexander-demicev @furkatgofurov7 @Danil-Grigorev @mjura
+
+## Context
+
+As an operator, I'd like to have an alternative import strategy that will generate a `management.cattle.io/v3` cluster instead of a `provisioning.cattle.io/v1` cluster. This is to ensure that we can support this scenario in light of future changes to Rancher.
+
+## Decision
+
+The current import controller logic creates a `provisioning.cattle.io/v1` cluster when a new CAPI cluster is detected. To support a new import strategy, based on a new feature flag, Rancher Turtles will either start the existing `provisioning.cattle.io` controller or a new one for `management.cattle.io`.
+
+Adding support for this resource presents some challenges:
+- Owner Reference cannot be set: `management.cattle.io/v3` clusters are cluster-scoped while CAPI clusters are namespaced.
```

Comment:
We can still set and ownership reference on the CAPI cluster, to ensure the deletion is followed one way at least.

Reply:
If I understand correctly, you mean setting the Rancher cluster as the owner and the CAPI cluster as the dependent object. If that's the case, wouldn't it conflict with the way we manage un-importing clusters? A user may want to remove a cluster from Rancher but keep the underlying CAPI cluster. I'm afraid setting owner references this way would break this functionality.

Reply:
No, it will not. When creating a v3 cluster for unmanaged scenario we should follow  https://kubernetes.io/docs/tasks/administer-cluster/use-cascading-deletion/#set-orphan-deletion-policy, and apply the `orphan` finalizer to the v3 cluster object. This way, when a regular delete is issued to v3 cluster, this will not follow with deletion of dependents, and will simply remove the ownership reference.

For managed scenario the finalizer can be omitted. If we go this route, we might not need labels too, as ownership reference contains the name of the v3 object to lookup.

Reply:
We’ve validated use of the ownership references in this scenario, unfortunately most of the CLI clients specify the default `background` policy during deletion, overriding the `orphan` policy on the resource, and this irreversibly causes capi cluster deletion. It’s better to use label matching, as it’s currently being suggested.